rust/protocol/src/lib.rs

```diff
@@ -3,9 +3,27 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Rust implementation of the **[Signal Protocol]** for asynchronous
+//! forward-secret public-key cryptography.
+//!
+//! In particular, this library implements operations conforming to the following specifications:
+//! - the **[X3DH]** key agreement protocol,
+//! - the **[Double Ratchet]** *(Axolotl)* messaging protocol,
+//! - the **[Sesame]** session agreement protocol.
+//!
+//! [Signal Protocol]: https://signal.org/
+//! [X3DH]: https://signal.org/docs/specifications/x3dh/
+//! [Double Ratchet]: https://signal.org/docs/specifications/doubleratchet/
+//! [Sesame]: https://signal.org/docs/specifications/sesame/
```

Comment:
I am still trying to come up with a simpler explanation of Sesame than what's in that spec, but it's about linking multiple devices to a single user, something that's not implemented in this library. We can drop it from the doc comment, at least for now.

Reply:
Thanks! That makes sense! I think I was confused at first mostly by the dizzying array of different IDs being handed around (which is addressed in #289) -- I assumed that something more complex had to be going on.